### ISSUE TITLE
Question about logitalpha

### ISSUE THREAD

**Issue by ZijiaLewisLu** [gngdb/variational-dropout#11]
Thank you for the code. I am trying to implement variational dropout in Pytorch. It is very helpful for me.

I just have one question. I am looking at layer for the "local reparameterization" paper. I saw to store alpha as logitalpha and sigmoid it to get alpha. I am wondering why it is so. If I am correct, logitalpha is never directly used, right?

**Comment by gngdb** [comment]
It's for optimisation. We want to have a value that is bounded between 0 and 1, but the optimiser doesn't know that, so if we just made a shared variable for alpha and updated it with stochastic gradients steps it would soon walk outside of 0 or 1. If we're only taking steps in logitalpha, then it can walk to negative or positive infinity and it doesn't matter because alpha *can never be below 0 or above 1*.

**Comment by ZijiaLewisLu** [comment]
Thank you!
